tornado-unittests/src/main/java/uk/ac/manchester/tornado/unittests/compiler/TestCompilerFlagsAPI.java

```diff
@@ -56,8 +56,38 @@ public void test() throws TornadoExecutionPlanException {
 
         try (TornadoExecutionPlan executionPlan = new TornadoExecutionPlan(taskGraph.snapshot())) {
             executionPlan.withCompilerFlags(TornadoVMBackendType.OPENCL, "-cl-opt-disable") //
-                    .withCompilerFlags(TornadoVMBackendType.PTX, " ") //
-                    .withCompilerFlags(TornadoVMBackendType.SPIRV, "-ze-opt-level 1") //
+                    .execute();
+        }
+
+    }
+
+    @Test
+    public void testPTX() throws TornadoExecutionPlanException {
+        FloatArray data = new FloatArray(512);
+        data.init(1.0f);
+
+        TaskGraph taskGraph = new TaskGraph("init") //
+                .task("foo", TestCompilerFlagsAPI::foo, data) //
+                .transferToHost(DataTransferMode.EVERY_EXECUTION, data);
+
+        try (TornadoExecutionPlan executionPlan = new TornadoExecutionPlan(taskGraph.snapshot())) {
+            executionPlan.withCompilerFlags(TornadoVMBackendType.PTX, "CU_JIT_OPTIMIZATION_LEVEL 0") //
+                    .execute();
+        }
+
+    }
+
+    @Test
+    public void testSPIRV() throws TornadoExecutionPlanException {
+        FloatArray data = new FloatArray(512);
+        data.init(1.0f);
+
+        TaskGraph taskGraph = new TaskGraph("init") //
+                .task("foo", TestCompilerFlagsAPI::foo, data) //
+                .transferToHost(DataTransferMode.EVERY_EXECUTION, data);
+
+        try (TornadoExecutionPlan executionPlan = new TornadoExecutionPlan(taskGraph.snapshot())) {
+            executionPlan.withCompilerFlags(TornadoVMBackendType.SPIRV, "-ze-opt-level 1") //
                     .execute();
```

Comment:
Regarding the tests, it is a bit confusing that testPTX() and testSPIR-V pass when we have only OpenCL.

I would suggest adding the `assertNotBackend` check in each of them so that the backends that are not built to not be tested.

In essence, each test to have one `asserNotBackend` call.